test/cppsuite/test_harness/workload_generator.h

```diff
@@ -96,67 +94,139 @@ class workload_generator {
+        std::this_thread::sleep_for(std::chrono::seconds(duration_seconds));
+        _thread_manager.finish();
+        debug_info("Run stage done", _trace_level, DEBUG_INFO);
+        return 0;
+    }
+
+    /* Workload threaded operations. */
+    static void
+    execute_operation(thread_context &context)
+    {
+        thread_operation operation;
+
+        operation = context.get_thread_operation();
+
+        if (context.get_session() == nullptr) {
+            testutil_die(DEBUG_ABORT, "system: execute_operation : Session is NULL");
+        }
+
+        switch (operation) {
+        case thread_operation::INSERT:
+            /* Sleep until it is implemented. */
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            break;
+        case thread_operation::READ:
+            read_operation(context);
+            break;
+        case thread_operation::REMOVE:
+            /* Sleep until it is implemented. */
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            break;
+        case thread_operation::UPDATE:
+            /* Sleep until it is implemented. */
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            break;
+        default:
+            testutil_die(DEBUG_ABORT, "system: thread_operation is unknown : %d",
+              static_cast<int>(thread_operation::UPDATE));
+            break;
+        }
+    }
+
+    /* Basic read operation that walks a cursors across all collections. */
+    static void
+    read_operation(thread_context &context)
+    {
+        WT_CURSOR *cursor;
+        std::vector<WT_CURSOR *> cursors;
+
+        /* Get a cursor for each collection in collection_names. */
+        for (const auto &it : context.get_collection_names()) {
+            testutil_check(context.get_session()->open_cursor(
+              context.get_session(), it.c_str(), NULL, NULL, &cursor));
+            cursors.push_back(cursor);
+        }
+
+        while (context.is_running()) {
+            /* Walk each cursor. */
+            for (const auto &it : cursors)
+                it->next(it);
+        }
     }
 
     /* WiredTiger APIs wrappers for single operations. */
-    int
-    insert(WT_CURSOR *cursor, const char *value)
+    static int
+    insert(WT_CURSOR *cursor)
     {
         if (cursor == nullptr)
             throw std::invalid_argument("Failed to call insert, invalid cursor");
         return (cursor->insert(cursor));
     }
 
-    int
+    static int
     search(WT_CURSOR *cursor)
     {
         if (cursor == nullptr)
             throw std::invalid_argument("Failed to call search, invalid cursor");
         return (cursor->search(cursor));
     }
 
-    int
+    static int
     search_near(WT_CURSOR *cursor, int *exact)
     {
         if (cursor == nullptr)
             throw std::invalid_argument("Failed to call search_near, invalid cursor");
         return (cursor->search_near(cursor, exact));
     }
 
-    int
+    static int
```

Comment:
Can we throw exception instead of return the errorcode?

Reply:
As discussed, we want to return the error code so the calling app can deal with it if required (i.e a test can expect a specific error code)